### DISCUSSION TITLE
How to use `return confirm('are you sure')` in an event handler

### DISCUSSION THREAD

**Discussion by tony84727** [alpinejs/alpine#1916] [1. Help] [answered]
Hello. I'm trying to use alpinejs v3 to implement a confirm dialog for a form.

I found this issue. https://github.com/alpinejs/alpine/issues/150

And the solution, PR #162, seems to  suggest that we can use
```html
<form x-data>
  <button @click="return confirm('are you sure')"
</form>
```
to do that.

But the above code will make alpinejs throws the `Error: Unexpected token 'return'` error.

Here's a snippet to reproduce the behavior.

https://stackblitz.com/edit/js-q1bdjm?file=index.html

What is the correct way in alpinejs (v3) to do `onclick="return confirm('are you sure')"`?

**Comment by SimoTod** [comment, marked as answer]
That PR was for Alpine v2.
V3 doesn't support that behaviour anymore, see https://alpinejs.dev/upgrade-guide#no-false-return-from-event-handlers

**Reply by SimoTod** [reply]
The closest thing you can do is 
```html
<form x-data>
   <button @click="if(!confirm('Are you sure?')) $event.preventDefault()">
  test</button> 
</form>
```

**Reply by tony84727** [reply]
Ah, I see. Thank you. That is exactly what I want.
